src/puppetlabs/puppetdb/scf/storage.clj

```diff
@@ -1359,6 +1359,20 @@
                                    reports/resources->resource-events
                                    (map normalize-resource-event)))))
 
+(defn filter-filter
+  [timestamp resource-events-ttl]
+  (let [after (time/after? (org.joda.time.DateTime. timestamp)
+               (.minus (now) resource-events-ttl))]
+    after)
+  )
+
+(defn filter-expired-resources
+   [resource-events-ttl resource-list]
+   (if (some? resource-events-ttl)
```

Comment:
I'm not sure if we need the `some?` check here

Reply:
I've noticed that in some tests `resource-events-ttl` is nil. Not sure wether it can happen in real life so just wanted to be sure it won't try to do unexpected operations on nil.